### ISSUE TITLE
UnhandledPromiseRejectionWarning: ValidationError

### ISSUE THREAD

**Issue by nikhilsindhu** [dynamoosejs/dynamoose#700]
### Summary:
I've created a User schema with a static method on it, to check if the current user (email address to be more specific) is unique or not. And I'm using Scan method to check that.

But when that static method is provided with an invalid Email address, It doesn't throw any validation error to catch block, but it throws an UnhandledPromiseRejectionWarning: ValidationError.

#### Schema
```
const UserSchema = new dynamoose.Schema(
  {
    test_id: {
      type: String,
      required: true,
      default: shortid.generate
    },
    name: {
      type: String,
      required: true,
      validate: function(val) {
        // only accept 150 characters or fewer
        return typeof val === 'string' && val.length <= 150 && val.length > 0
      }
    },
    email: {
      type: String,
      required: true,
      lowercase: true,
      validate: /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
    },
    password: {
      required: true,
      type: String,
      isAsync: true,
      set: async val => {
        try {
          const salt = await bcrypt.genSalt(10)
          const securePassword = await bcrypt.hash(val, sal)
          return securePassword
        } catch (error) {
          console.error('Error :', error)
          throw new ErrorResponse('Password Encryption failed!', 500)
        }
      }
    },
    reset_password_token: {
      required: false,
      type: String,
      default: null
    },
    reset_password_expire: {
      required: false,
      type: Number,
      default: null
    },
    is_active: {
      required: false,
      type: Boolean,
      default: true
    },
    is_superuser: {
      required: false,
      type: Boolean,
      default: false
    },
    user_permissions: {
      required: false,
      type: [String],
      default: []
    }
  },
  {
    timestamps: {
      createdAt: 'date_joined',
      updatedAt: 'date_updated'
    }
  }
)
```

#### Model
```
UserSchema.statics.findIfUniqueUser = function({ email : "abc.com" }) {
  return new Promise(resolve => {
    try{
      this.scan('email')
        .contains(email)
        .exec((error, users) => {
          if (error) {
            console.error('Error :', error)
            const errorMsg = error || 'Server error!'
            throw new ErrorResponse(errorMsg, 500)
          }
          resolve(Array.isArray(users) && users.length === 0 ? true : false)
        })
    } catch(error){
      console.log({error})
    }
  })
}

const User = dynamoose.model('user', UserSchema)
```

#### General
```
  try {
    const result = await User.findIfUniqueUser({ email : "invalid_email_at.com"})  // passing an Invalid email address
    console.log({
      result
    })
  } catch (error) {
    console.log({ error })
  }
```

### Current output and behavior:

UnhandledPromiseRejectionWarning: ValidationError: Validation failed: email
    at Attribute.toDynamo (D:\Dev Shop\Betaflux\Backend\ultraviolette\uv-admin-dashboard\node_modules\dynamoose\dist\Attribute.js:378:15)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at Scan.exec (D:\Dev Shop\Betaflux\Backend\ultraviolette\uv-admin-dashboard\node_modules\dynamoose\dist\Scan.js:118:178)
(node:11448) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)

### Expected output and behaviour:

Should throw an error, that can be caught in try-catch block.

**Comment by fishcharlie** [comment]
@nikhilsindhu What are you trying to do with `function({ email : "abc.com" }) `? That object within the function parameters looks really wrong to me.

I'm curious if your exec function even gets run.

It looks like it's coming from this line tho:

https://github.com/dynamoosejs/dynamoose/blob/32c65fe5bc652b28abc896cc9051ca84d41d0622/lib/Attribute.js#L405

I can't really think of a way to get around this off the top of my head other than using the AWS-SDK directly to run your scan.

---

This is the type of things I'm hoping the v2.0 rewrite (#677) will fix. There are a lot of these edge cases, where Dynamoose is trying to be super strict, but there are situations where it doesn't make sense. And the strictness leads to cases not working correctly, or weird edge cases happening.

Overall I'm very happy with the direction of the v2.0 rewrite so far. But I'd love some more community feedback on it. I just started working on the system for default values, and next will be validation. But the system will have certain similarities. I think the system I designed has enough flexibility to cover these type of edge cases, but I'd love some feedback on it. Hopefully I'll be able to push those changes soon.

If you'd like, I'd love some feedback on it. Even just leaving a comment on that PR with a link to this thread would be great. It'd be even better if you'd write some tests around this functionality to ensure we cover your use cases correctly.

---

All that being said. There should be some way to catch this error and handle it correctly. But it almost feels like Dynamoose might not be sending that error up the chain to your code all the way. And I'm not really sure how to fix that without playing around with it for a while.
